### PR TITLE
Staging for K8S testing

### PR DESCRIPTION
**Pull request by kaikawaliu**
## Description

This is a summative PR for recent several PRs:
https://github.com/Zilliqa/Zilliqa/pull/440
https://github.com/Zilliqa/Zilliqa/pull/442
https://github.com/Zilliqa/Zilliqa/pull/444
https://github.com/Zilliqa/Zilliqa/pull/451
https://github.com/Zilliqa/Zilliqa/pull/504
https://github.com/Zilliqa/Zilliqa/pull/509
https://github.com/Zilliqa/Zilliqa/pull/530

## Review Suggestion

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test (200 nodes test with txn dispatcher)

### CODE REVIEW

File: constants.xml [hunk 1]
```diff
@@ -18,23 +18,24 @@
         <CONSENSUS_OBJECT_TIMEOUT>10</CONSENSUS_OBJECT_TIMEOUT>
         <FETCHING_MISSING_TXNS_TIMEOUT>10</FETCHING_MISSING_TXNS_TIMEOUT>
         <FINALBLOCK_CONSENSUS_OBJECT_TIMEOUT>30</FINALBLOCK_CONSENSUS_OBJECT_TIMEOUT>
-        <NUM_FINAL_BLOCK_PER_POW>50</NUM_FINAL_BLOCK_PER_POW>
+        <NUM_FINAL_BLOCK_PER_POW>5</NUM_FINAL_BLOCK_PER_POW>
```
Comment (bb111189) [inline constants.xml]: should be retain as 50

File: src/libNode/MicroBlockProcessing.cpp [hunk 2]
```diff
@@ -333,35 +397,25 @@ bool Node::ComposeMicroBlock()
...
 
-        for (const auto& tx : submittedTransactions)
+        numTxs = processedTransactions.size();
+        if (numTxs != m_TxnOrder.size())
         {
-            const auto& txid = tx.first.asArray();
-            copy(txid.begin(), txid.end(),
-                 tranHashes.at(index).asArray().begin());
-            index++;
+            LOG_GENERAL(FATAL,
```
Comment (KaustubhShamshery) [inline src/libNode/MicroBlockProcessing.cpp]: [Antonio] I'm thinking if we should make it FATAL, or attempt to rejoin instead

File: src/cmd/Gen_Test_Transaction.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                            {});
+            /*txns.emplace_back(txn);*/
+            txn.Serialize(txns, 0);
+            for (auto& i : txns)
+            {
+                file << i;
+            }
+        }
+
+        file.close();
```
Comment (ckyang) [inline src/cmd/Gen_Test_Transaction.cpp]: fstream(fileName, mode) will automatically call open()/close() inside constructor/destructor. Suggest to use this pattern to enhance readability and maybe we can improve in follow-up PR.

File: src/depends/common/FixedHash.h [hunk 4]
```diff
@@ -43,7 +43,7 @@ namespace dev
     /// Transparently converts to/from the corresponding arithmetic type; this will
     /// assume the data contained in the hash is big-endian.
     template <unsigned N>
-    class FixedHash
+    class FixedHash //: boost::less_than_comparable<FixedHash<N>>
```
Comment (ckyang) [inline src/depends/common/FixedHash.h]: Remove the redundant comment

File: src/libData/AccountData/AccountStore.cpp [hunk 5]
```diff
@@ -95,7 +95,7 @@ int AccountStore::Deserialize(const vector<unsigned char>& src,
             UpdateStateTrie(address, account);
             // MoveUpdatesToDisk();
         }
-        PrintAccountState();
+        // PrintAccountState();
```
Comment (ckyang) [inline src/libData/AccountData/AccountStore.cpp]: Remove the redundant comment

File: src/libData/AccountData/AccountStore.cpp [hunk 6]
```diff
@@ -335,6 +339,20 @@ bool AccountStore::UpdateCoinbaseTemp(const Address& rewardee,
     //Should the nonce increase ??
 }
 
+boost::multiprecision::uint256_t
+AccountStore::GetNonceTemp(const Address& address)
+{
+    auto it = m_accountStoreTemp->GetAddressToAccount()->find(address);
+    if (it != m_accountStoreTemp->GetAddressToAccount()->end())
```
Comment (ckyang) [inline src/libData/AccountData/AccountStore.cpp]: Directly compare find()/end(), to reduce the usage of extra variable "it"

File: constants_local.xml [hunk 7]
```diff
@@ -61,7 +62,11 @@
         <OUTPUT_JSON>output.json</OUTPUT_JSON>
         <INPUT_CODE>input.scilla</INPUT_CODE>
     </smart_contract>
-    <accounts>
+	<TransactionDispatcher>
```
Comment (Gnnng) [inline constants_local.xml]: Some format issue

File: constants_local.xml [hunk 8]
```diff
@@ -61,7 +62,11 @@
...
+        <USE_REMOTE_TXN_CREATOR>false</USE_REMOTE_TXN_CREATOR>
+        <TXN_PATH></TXN_PATH>
+    </TransactionDispatcher>
+   <accounts>
```
Comment (Gnnng) [inline constants_local.xml]: Format issue

File: constants.xml [hunk 9]
```diff
@@ -48,6 +49,10 @@
         <OPENCL_GPU_MINE>false</OPENCL_GPU_MINE>
         <CUDA_GPU_MINE>false</CUDA_GPU_MINE>
     </options>
+   <TransactionDispatcher>
```
Comment (Gnnng) [inline constants.xml]: Format issue

File: src/libData/DataStructures/MultiIndexContainer.h [hunk 10]
```diff
@@ -0,0 +1,71 @@
+/**
```
Comment (KaustubhShamshery) [inline src/libData/DataStructures/MultiIndexContainer.h]: Same file is in src/common ?

File: src/libDirectoryService/FinalBlockPreProcessing.cpp [hunk 11]
```diff
@@ -84,6 +79,7 @@ void DirectoryService::ExtractDataFromMicroblocks(
             && microBlock.GetHeader().GetStateDeltaHash() == StateHash();
 
         bool isEmptyTxn = (microBlock.GetHeader().GetNumTxs() == 0);
+        //[CNBSE]
```
Comment (KaustubhShamshery) [inline src/libDirectoryService/FinalBlockPreProcessing.cpp]: Redundant

File: constants.xml [hunk 12]
```diff
@@ -48,6 +49,10 @@
         <OPENCL_GPU_MINE>false</OPENCL_GPU_MINE>
         <CUDA_GPU_MINE>false</CUDA_GPU_MINE>
     </options>
+    <TransactionDispatcher>
+        <USE_REMOTE_TXN_CREATOR>false</USE_REMOTE_TXN_CREATOR>
```
[the remark this answers is not in the document]
Comment (kaikawaliu) [reply constants.xml]: Noted, will do this in the next PR.

[the remark this answers is not in the document]
Comment (Gnnng) [reply constants.xml]: This option is useful to me to select the lookup for dispatching (see https://github.com/Zilliqa/Zilliqa/commit/332897f014d3bc62ece16a49cac964cad3896410) so we can keep it

File: src/libData/DataStructures/MultiIndexContainer.h (hunk 10 above)
Comment (kaikawaliu) [reply src/libData/DataStructures/MultiIndexContainer.h]: Noted, will remove in the next PR

File: src/libDirectoryService/FinalBlockPreProcessing.cpp (hunk 11 above)
Comment (kaikawaliu) [reply src/libDirectoryService/FinalBlockPreProcessing.cpp]: Noted, will remove in the next PR

File: src/libData/AccountData/AccountStore.cpp (hunk 6 above)
Comment (kaikawaliu) [reply src/libData/AccountData/AccountStore.cpp]: Noted

**Pull request merged**